moveit_ros/planning/kinematics_plugin_loader/src/kinematics_plugin_loader.cpp

```diff
@@ -68,16 +69,17 @@ class KinematicsPluginLoader::KinematicsLoaderImpl
    * \param search_res
    * \param iksolver_to_tip_links - a map between each ik solver and a vector of custom-specified tip link(s)
    */
-  KinematicsLoaderImpl(const rclcpp::Node::SharedPtr& node, const std::string& robot_description,
+  KinematicsLoaderImpl(node_interface::NodeInterfaceSharedPtr node_interface, const std::string& robot_description,
                        const std::map<std::string, std::vector<std::string>>& possible_kinematics_solvers,
                        const std::map<std::string, std::vector<double>>& search_res,
                        const std::map<std::string, std::vector<std::string>>& iksolver_to_tip_links)
-    : node_(node)
+    : node_interface_{ std::move(node_interface) }
```

Comment:
Please use this interface style in all the other constructors that have shared_ptr parameters.